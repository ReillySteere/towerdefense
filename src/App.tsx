import React from 'react';
import { useQuery } from '@tanstack/react-query';
import styles from './styles.module.scss';
import './phaserGame';
import { useGameStore } from './store';

const fetchGreeting = async (): Promise<string> => {
  // Simulate an API call with a delay.
  return new Promise((resolve) =>
    setTimeout(() => resolve('Hello from the API!'), 1000),
  );
};

const App = () => {
  const { score, incrementScore } = useGameStore();
  const { data, isLoading, error } = useQuery({
    queryKey: ['greeting'],
    queryFn: fetchGreeting,
  });

  return (
    <div className={styles.container}>
      <div id="phaser-container" className={styles.phaserContainer}></div>
      <div id="root-ui" className={styles.uiOverlay}>
        <h1>React UI Overlay</h1>
        <button onClick={incrementScore}>Increment Score</button>
        <p>
          {isLoading ? (
            <span>'Loading greeting...'</span>
          ) : error ? (
            <span>'Error fetching greeting'</span>
          ) : (
            <span>{JSON.stringify(data)}</span>
          )}
        </p>
      </div>
      <p>Score: {score}</p>
    </div>
  );
};

export default App;
