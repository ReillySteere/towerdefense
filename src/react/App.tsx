import React from 'react';
import { useQuery } from '@tanstack/react-query';
import styles from './styles.module.scss';
import { useGameStore } from './store';
import { fetchGreeting } from '../api';

const App = () => {
  const { score, incrementScore } = useGameStore();
  const { data, isLoading, error } = useQuery({
    queryKey: ['greeting'],
    queryFn: fetchGreeting,
  });

  return (
    <div className={styles.container}>
      <div id="phaser-container" className={styles.phaserContainer}></div>
      <div id="root-ui" className={styles.uiOverlay}>
        <h1>React UI Overlay</h1>
        <button onClick={incrementScore}>Increment Score</button>
        <p>
          {isLoading ? (
            <span>'Loading greeting...'</span>
          ) : error ? (
            <span>'Error fetching greeting'</span>
          ) : (
            <span>{JSON.stringify(data)}</span>
          )}
        </p>
      </div>
      <p>Score: {score}</p>
    </div>
  );
};

export default App;
