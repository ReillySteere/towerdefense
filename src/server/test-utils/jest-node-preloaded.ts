// @ts-expect-error - setup script
global.self = {};

global.TextEncoder = TextEncoder;
global.TextDecoder = TextDecoder;

afterAll(() => {
  jest.clearAllMocks();
});
