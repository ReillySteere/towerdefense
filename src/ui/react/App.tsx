import React from 'react';
import styles from './styles.module.scss';
import HealthStatus from './components/HealthStatus';

const App = () => {
  return (
    <div className={styles.container}>
      <div id="phaser-container" className={styles.phaserContainer}></div>
      <div id="root-ui" className={styles.uiOverlay}>
        <HealthStatus />
      </div>
    </div>
  );
};

export default App;
