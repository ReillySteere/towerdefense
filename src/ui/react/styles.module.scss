$bg-color: #24252a;
$border-color: #444;

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #222;
  min-height: 100vh;
  padding: 20px;
}

.phaserContainer {
  border: 2px solid $border-color;
  margin-bottom: 20px;
  width: 1400px;
  height: 700px;
}

.uiOverlay {
  width: 1400;
  text-align: center;
}
