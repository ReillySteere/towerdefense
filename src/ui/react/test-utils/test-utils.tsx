import React from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, RenderOptions } from '@testing-library/react';
import type { ReactElement } from 'react';

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });

const AllTheProviders = ({ children }: { children?: React.ReactNode }) => {
  const testQueryClient = createTestQueryClient();
  return (
    <QueryClientProvider client={testQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
