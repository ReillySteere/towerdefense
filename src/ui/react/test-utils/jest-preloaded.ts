import '@testing-library/jest-dom';

import { enableFetchMocks } from 'jest-fetch-mock';
import { cleanup } from './test-utils';

/* eslint camelcase: "off" */
(global as any).__non_webpack_require__ = (path: string) => require(path);
(global as any).__webpack_public_path__ = '';

enableFetchMocks();

afterEach(cleanup);
