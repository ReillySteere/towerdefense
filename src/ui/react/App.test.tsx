import React from 'react';
import { render, screen } from 'ui/react/test-utils';
import App from './App';

// Updated jest.mock to avoid referencing out-of-scope React
jest.mock('./components/HealthStatus', () => {
  const React = require('react');
  return {
    __esModule: true,
    default: function HealthStatusMock() {
      return React.createElement(
        'div',
        { 'data-testid': 'health-status' },
        'Health Status Mock',
      );
    },
  };
});

describe('App Component', () => {
  it('renders the main container with correct child elements', () => {
    const { container } = render(<App />);
    const mainContainer = container.querySelector('div');
    expect(mainContainer).toBeInTheDocument();
  });

  it('renders the phaser container with proper class', () => {
    const { container } = render(<App />);
    const phaserContainer = container.querySelector('#phaser-container');
    expect(phaserContainer).toBeInTheDocument();
    expect(phaserContainer).toHaveClass('phaserContainer');
  });

  it('renders the UI overlay container with proper class', () => {
    const { container } = render(<App />);
    const uiOverlay = container.querySelector('#root-ui');
    expect(uiOverlay).toBeInTheDocument();
    expect(uiOverlay).toHaveClass('uiOverlay');
  });

  it('renders the HealthStatus component', () => {
    render(<App />);
    const healthStatus = screen.getByTestId('health-status');
    expect(healthStatus).toBeInTheDocument();
    expect(healthStatus).toHaveTextContent('Health Status Mock');
  });

  it('matches snapshot', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('applies the proper container class on the root element', () => {
    const { container } = render(<App />);
    const rootElement = container.firstChild;
    expect(rootElement).toHaveClass('container');
  });
});
