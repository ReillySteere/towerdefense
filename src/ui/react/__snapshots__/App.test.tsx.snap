// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div
  class="container"
>
  <div
    class="phaserContainer"
    id="phaser-container"
  />
  <div
    class="uiOverlay"
    id="root-ui"
  >
    <div
      data-testid="health-status"
    >
      Health Status Mock
    </div>
  </div>
</div>
`;
